### PR TITLE
vcmi: init at 1.0.0

### PR DESCRIPTION
**Pull request by azahi**
###### Description of changes

Open-source engine for HoMM3.

https://vcmi.eu/
https://github.com/vcmi/vcmi

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/vcmi/default.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    ffmpeg
+    luajit
+    qtbase
+    tbb
+    zlib
+  ];
+
+  postPatch = ''
+    substituteInPlace Version.cpp.in \
+      --replace "@GIT_SHA1@" "${src.rev}";
```
Comment (sikmir) [inline pkgs/games/vcmi/default.nix:59]:
```suggestion
      --subst-var-by GIT_SHA1 "${src.rev}";
```

File: pkgs/games/vcmi/default.nix [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    ffmpeg
+    luajit
+    qtbase
+    tbb
+    zlib
+  ];
+
+  postPatch = ''
+    substituteInPlace Version.cpp.in \
+      --subst-var-by GIT_SHA1 "${src.rev}";
```
Comment (trofi) [inline pkgs/games/vcmi/default.nix:59]: When I get past main menu and run actual scenario the program fails to load it's libraries:

```
Cannot open dynamic library (/nix/store/gnjm0qjp69bndm9dpilmc0zk8kgkcrmg-vcmi-unstable-2022-09-11//nix/store/gnjm0qjp69bndm9dpilmc0zk8kgkcrmg-vcmi-unstable-2022-09-11/lib/vcmi/AI/libVCAI.so). Throwing...
terminate called after throwing an instance of 'std::runtime_error'
  what():  Cannot open dynamic library
```

Locally I have been working it around with

```
  cmakeFlags = [
    # Make libvcmi.so discoverable in non-standard location.
    # Default rpath does not contain it at all.
    "-DCMAKE_INSTALL_RPATH=${placeholder "out"}/lib/vcmi"

    "-DENABLE_PCH=OFF"

    # Upstream assumes relative value whilc nixpkgs passes absolute.
    # Both should be allowed:
    #  https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html
    # Meawhile work it around by passing relative value.
    "-DCMAKE_INSTALL_LIBDIR=lib"
  ];
```

**Review by trofi — CHANGES_REQUESTED** [review]

File: pkgs/games/vcmi/default.nix (hunk 2 above)
Comment (azahi) [reply pkgs/games/vcmi/default.nix]: Thanks, I've added these.

**Review by trofi — APPROVED** [review]
Looks good. One question: what is `autoPatchelfHook` used for?

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 189729` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vcmi</li>
  </ul>
</details>

**Comment by azahi** [comment]
> Looks good. One question: what is `autoPatchelfHook` used for?

I've used it to fix RPATHs for libraries in `lib`. Not sure if this is still needed now that we specify the cmake flag. Let me check that.

**Comment by azahi** [comment]
Alright, removed it. I've also disabled building tests since we don't run checkPhase (because sandboxed tests for this package are kinda broken).

**Comment by trofi** [comment]
Pulled as is. Thank you!

**Pull request merged**